stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -467,25 +466,21 @@ public class ChatClient internal constructor(
         offset: Int,
         limit: Int,
         type: String
-    ): Call<List<AttachmentWithDate>> {
+    ): Call<List<Attachment>> =
+        getMessagesWithAttachments(channelType, channelId, offset, limit, type).map { messages ->
+            messages.flatMap { message -> message.attachments }
+        }
+
+    public fun getMessagesWithAttachments(
+        channelType: String,
+        channelId: String,
+        offset: Int,
+        limit: Int,
+        type: String
+    ): Call<List<Message>> {
         val channelFilter = Filters.`in`("cid", "$channelType:$channelId")
         val messageFilter = Filters.`in`("attachments.type", type)
-
-        return searchMessages(SearchMessagesRequest(offset, limit, channelFilter, messageFilter)).map { messages ->
-            messages.flatMap { message ->
-                if (type == ATTACHMENT_TYPE_IMAGE) {
-                    message.attachments.filter { attachment -> attachment.isImage }
```

Comment:
Are we sure this filtering redundant in the previous version of the code? We added it quite recently, in https://github.com/GetStream/stream-chat-android/commit/9b7778d4ebd82a2bea78233163ff2e36dd5f347b. 

@adasiewiczr got any insight on this one?

Reply:
I guess this https://github.com/GetStream/stream-chat-android/pull/1198#discussion_r556338320 is the answer 😃

Reply:
exactly :)